genesis/ext/pyrender/interaction/viewer_interaction.py

```diff
@@ -197,53 +201,62 @@ def _raycast_against_ground_plane(self, ray: Ray) -> RayHit:
 
     def raycast_against_entity_obb(self, entity: RigidEntity, ray: Ray) -> RayHit:
         if isinstance(entity.morph, gs.morphs.Box):
-            box: gs.morphs.Box = entity.morph
-            size = Vec3.from_xyz(*box.size)
-            pose = self.get_pose_of_first_geom(entity)
-            aabb = AABB.from_center_and_size(Vec3.zero(), size)
-            ray_hit = aabb.raycast_obb(pose, ray)
+            obb: OBB = self._get_box_obb(entity)
+            ray_hit = obb.raycast(ray)
+            if ray_hit.is_hit:
+                ray_hit.geom = entity.geoms[0]
             return ray_hit
-        else:
+        elif isinstance(entity.morph, gs.morphs.Plane):
+            # ignore plane
             return RayHit.no_hit()
-
-    def raycast_against_entities(self, ray: Ray) -> tuple[RayHit | None, RigidEntity | None]:
-        closest_hit = None
-        hit_entity: RigidEntity | None = None
-        for entity in self.get_entities():
-            ray_hit = self.raycast_against_entity_obb(entity, ray)
-            if ray_hit.is_hit and (closest_hit is None or ray_hit.distance < closest_hit.distance):
+        else:
+            closest_hit = RayHit.no_hit()
+            for link in entity.links:
+                # how to check if the link is dynamic ??
+                if 0.0 < link.get_mass(): 
+                    for geom in link.geoms:
+                        obb: OBB = self._get_geom_placeholder_obb(geom)
+                        ray_hit = obb.raycast(ray)
+                        if ray_hit.distance < closest_hit.distance:
+                            ray_hit.geom = geom
+                            closest_hit = ray_hit
+            return closest_hit
+
+    def raycast_against_entities(self, ray: Ray) -> RayHit:
+        closest_hit = RayHit.no_hit()
+        for entity in self.scene.sim.rigid_solver.entities:
+            rigid_entity: RigidEntity = cast(RigidEntity, entity)
+            ray_hit = self.raycast_against_entity_obb(rigid_entity, ray)
+            if ray_hit.distance < closest_hit.distance:
                 closest_hit = ray_hit
-                hit_entity = entity
-        return (closest_hit, hit_entity)
+        return closest_hit
 
-    def get_entities(self) -> list[RigidEntity]:
-        return self.scene.sim.rigid_solver.entities
+    def _get_box_obb(self, box_entity: RigidEntity) -> OBB:
+        box: gs.morphs.Box = box_entity.morph
+        pose = Pose.from_link(box_entity.links[0])
+        size = Vec3.from_xyz(*box.size)
+        return OBB.from_center_and_size(size, pose)
 
-    def get_pose_of_first_geom(self, entity: RigidEntity) -> 'Pose':
-        return Pose.from_geom(entity.geoms[0])
+    def _get_geom_placeholder_obb(self, geom: 'RigidGeom') -> OBB:
+        pose = Pose.from_geom(geom)
+        size = 0.125 * Vec3.one()
```

Comment:
Could you add `def full(fill_value):` ? That would be more convenient and efficient.